### PR TITLE
rnnoise: simplify initialization

### PR DESCRIPTION
**Pull request by saghul**
Depends on https://github.com/jitsi/rnnoise-wasm/pull/2

### CODE REVIEW

File: react/features/stream-effects/rnnoise/index.js [hunk 1]
```diff
@@ -9,29 +9,17 @@ import RnnoiseProcessor from './RnnoiseProcessor';
...
-                onRuntimeInitialized() {
-                    resolve();
-                },
-                onAbort(reason) {
-                    reject(reason);
-                }
-            });
-        });
+    if (!rnnoiseModule) {
+        rnnoiseModule = rnnoiseWasmInit();
```
Comment (andrei-gavrilescu) [inline react/features/stream-effects/rnnoise/index.js:21]: I remembered having some issues with the rnnoise promise based init (that's why all the nice boilerplate). I assume you tested with an image build using a newer version of emscripten.
This really looks much better.

Comment (saghul) [reply react/features/stream-effects/rnnoise/index.js:21]: Indeed!

**Comment by saghul** [comment]
Updated PTAL @andrei-gavrilescu

**Comment by ishmum123** [comment]
Hi @andrei-gavrilescu is it possible to use the 'noise-suppression' mechanism for rnnoise? If so, could you point me towards any direction?

**Comment by andrei-gavrilescu** [comment]
> Hi @andrei-gavrilescu is it possible to use the 'noise-suppression' mechanism for rnnoise? If so, could you point me towards any direction?

Hello there,

Yes you can, however you'd have to make some small adjustments; In the `RnnoiseProcessor` you have this function :`calculateAudioFrameVAD(pcmFrame: Float32Array)`  which has the following return `return this._wasmInterface._rnnoise_process_frame(this._context, this._wasmPcmOutput, this._wasmPcmInput);`
Now ` this._wasmPcmOutput` should, in theory, be the denoised input frame. Currently it's ignored and we just return the VAD score but that's what you're after.

**Comment by damienlaine** [comment]
Hello @andrei-gavrilescu
I'm also trying to fetch the denoised audio frame. this._wasmPcmOutput is only a pointer, it contains no data by itself.
The this._wasmInterface._rnnoise_process_frame(this._context,` this._wasmPcmOutput, this._wasmPcmInput); function works well but i can't find a way to fetch the denoised frame itself.

Many thanks for this useful work,

**Pull request merged**